Conflicting `BYTE_ORDER` definitions

When building `HttpServer_FirmwareUpload` sample with esp-quick-toolchain, a `"BYTE_ORDER" redefined` error is raised. CI builds samples with `STRICT=1` which converts this to a warning, however this is unsafe as the definitions are incompatible.

First definition is in `esp-quick-toolchain/xtensa-lx106-elf/xtensa-lx106-elf/include/machine/endian.h`
Conflicting definition is in `Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/arch/cc.h`

I'm raising this as an issue as the correct solution requires some thought and may cause other issues.

Original error from build logs:

```
/home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/arch/cc.h:65: warning: "BYTE_ORDER" redefined
   65 | #define BYTE_ORDER LITTLE_ENDIAN
      | 
In file included from /home/appveyor/projects/sming/opt/esp-quick-toolchain/xtensa-lx106-elf/xtensa-lx106-elf/include/sys/types.h:49,
                 from /home/appveyor/projects/sming/opt/esp-quick-toolchain/xtensa-lx106-elf/xtensa-lx106-elf/include/stdio.h:61,
                 from /home/appveyor/projects/sming/Sming/Wiring/FakePgmSpace.h:13,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/arch/cc.h:43,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/lwip/arch.h:43,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/lwip/debug.h:35,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/lwip/opt.h:46,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip/include/lwip/init.h:35,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp8266/include/espinc/lwip_includes.h:11,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/System/include/user_config.h:13,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp8266/ESP8266_NONOS_SDK/include/osapi.h:30,
                 from /home/appveyor/projects/sming/Sming/Arch/Esp8266/Components/esp8266/include/esp_systemapi.h:40,
                 from /home/appveyor/projects/sming/Sming/Libraries/libsodium/libsodium/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c:1:
/home/appveyor/projects/sming/opt/esp-quick-toolchain/xtensa-lx106-elf/xtensa-lx106-elf/include/machine/endian.h:27: note: this is the location of the previous definition
   27 | #define BYTE_ORDER _BYTE_ORDER
      | 
AR /home/appveyor/projects/sming/Sming/out/Esp8266/debug/lib/clib-libsodium.a
```

--- a/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip.patch
+++ b/Sming/Arch/Esp8266/Components/esp-open-lwip/esp-open-lwip.patch
@@ -76,10 +76,10 @@ index 1bc584f..e4af916 100644
  $(LIB): $(OBJS)
  	$(AR) rcs $@ $^
 diff --git a/include/arch/cc.h b/include/arch/cc.h
-index ff03b30..5efcf13 100644
+index ff03b30..4a2b4d9 100644
 --- a/include/arch/cc.h
 +++ b/include/arch/cc.h
-@@ -38,8 +38,27 @@
+@@ -38,14 +38,31 @@
  #include "c_types.h"
  #include "ets_sys.h"
  #include "osapi.h"
@@ -106,8 +106,15 @@ index ff03b30..5efcf13 100644
 +
  //#define LWIP_PROVIDE_ERRNO
  
- #if (1)
-@@ -56,6 +75,7 @@ typedef signed     short   s16_t;
+-#if (1)
++#ifndef BYTE_ORDER
+ #define BYTE_ORDER LITTLE_ENDIAN
+-#else
+-#define BYTE_ORDER BIG_ENDIAN
+ #endif
+ 
+ 
+@@ -56,6 +73,7 @@ typedef signed     short   s16_t;
  typedef unsigned   long    u32_t;
  typedef signed     long    s32_t;
  typedef unsigned long   mem_ptr_t;
@@ -115,7 +122,7 @@ index ff03b30..5efcf13 100644
  
  #define S16_F "d"
  #define U16_F "d"
-@@ -73,11 +93,12 @@ typedef unsigned long   mem_ptr_t;
+@@ -73,11 +91,12 @@ typedef unsigned long   mem_ptr_t;
  #define PACK_STRUCT_BEGIN
  #define PACK_STRUCT_END
  

